src/Presentation/Nop.Web/Controllers/OrderController.cs

```diff
@@ -207,7 +211,11 @@ public virtual async Task<IActionResult> ReOrder(int orderId)
             if (order == null || order.Deleted || customer.Id != order.CustomerId)
                 return Challenge();
 
-            await _orderProcessingService.ReOrderAsync(order);
+            var warnings = await _orderProcessingService.ReOrderAsync(order);
+
+            if (warnings.Any())
+                _notificationService.WarningNotification("Some products are not available anymore, so they weren't added to the cart.");

```

Comment:
@mike-warren-dev You should add a locale resource to allow translation of this warning text.